ARMC5 build of a mbed-os-example-blinky broken on Mbed OS master

### Description

ARMC5 build of a mbed-os-example-blinky is broken on Mbed OS master. Reverting C++11 enablement from profiles fixes the errors.

Test environment:
* ARM Compiler 5.06 update 6 (build 750)
* Mbed OS 6a1ab73988e87ccae28ebb6f944a7d448e86d9d7
* mbed-os-example-blinky 880ed00bff6352cea56031d34b9d891ba21059e9
* target: K64F
* profile: release

First error:
```
mbed compile -m K64F -t ARMC5 --profile=release 
<..>
Building project mbed-os-example-blinky (K64F, ARMC5)
Scan: mbed-os-example-blinky
Compile [ 69.8%]: at24mac.cpp
[Error] SingletonPtr.h@136,0:  #260-D: explicit type is missing ("int" assumed)
[Error] SingletonPtr.h@136,0:  #65: expected a ";"
[ERROR] "./mbed-os/platform/SingletonPtr.h", line 136: Error:  #260-D: explicit type is missing ("int" assumed)
"./mbed-os/platform/SingletonPtr.h", line 136: Error:  #65: expected a ";"
./mbed-os/components/802.15.4_RF/atmel-rf-driver/source/at24mac.cpp: 0 warnings, 2 errors
```

fix:
```
diff --git c/platform/SingletonPtr.h w/platform/SingletonPtr.h
index 9d2cc66..982f8ea 100644
--- c/platform/SingletonPtr.h
+++ w/platform/SingletonPtr.h
@@ -131,7 +131,9 @@ struct SingletonPtr {
 
     // This is zero initialized when in global scope
     mutable void *_ptr;
-#if __cplusplus >= 201103L
+
+    // ARMC5 does not support alignas()
+#if (__cplusplus >= 201103L) && !(defined(__ARMCC_VERSION) && (__ARMCC_VERSION < 6120000))
     // Align data appropriately
     alignas(T) mutable char _data[sizeof(T)];
 #else
```

next error:
```
Compile [ 87.4%]: NetworkInterface.cpp
[Error] NetworkInterface.cpp@101,0:  #20: identifier "UINT_FAST8_MAX" is undefined
[ERROR] "./mbed-os/features/netsocket/NetworkInterface.cpp", line 101: Error:  #20: identifier "UINT_FAST8_MAX" is undefined
./mbed-os/features/netsocket/NetworkInterface.cpp: 0 warnings, 1 error
```

### Issue request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the issue otherwise.)
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Question
    [ ] Enhancement
    [x] Bug

--- a/features/netsocket/NetworkInterface.cpp
+++ b/features/netsocket/NetworkInterface.cpp
@@ -14,12 +14,12 @@
  * limitations under the License.
  */
 
+#include "ns_list.h"
 #include "netsocket/NetworkInterface.h"
 #include "netsocket/NetworkStack.h"
 #include "platform/Callback.h"
 #include "platform/mbed_error.h"
 #include <string.h>
-#include "ns_list.h"
 
 // Default network-interface state
 void NetworkInterface::set_as_default()

--- a/platform/SingletonPtr.h
+++ b/platform/SingletonPtr.h
@@ -131,8 +131,8 @@ struct SingletonPtr {
 
     // This is zero initialized when in global scope
     mutable void *_ptr;
-#if __cplusplus >= 201103L
-    // Align data appropriately
+#if __cplusplus >= 201103L && !defined __CC_ARM
+    // Align data appropriately (ARM Compiler 5 does not support alignas in C++11 mode)
     alignas(T) mutable char _data[sizeof(T)];
 #else
     // Force data to be 8 byte aligned

